### PR TITLE
[flutter_tools] support files in flutter analyze #96231

### LINKED ISSUE #96231: `flutter analyze` should accept single dart files as parameter
`flutter analyze` takes quite a while on a large code base. It would be helpful (e.g. for a Git pre-commit hook)
if it would be possible to specify the files, which should be checked. for example:

`flutter analyze lib/some_file.dart lib/some_other_file.dart` 

BTW, this is already implemented for `dart analyze`.

### PR DESCRIPTION
**Pull request by Jasguerrero**
https://github.com/flutter/flutter/issues/96231
"analyze" command only accepted directories now you can pass files to it

### CODE REVIEW

**Comment by christopherfujino** [comment]
@Jasguerrero you will have to rebase upstream in order to get the updated names of the `tool_integration_tests`. This is a breaking change I made to the infrastructure unrelated to your test. Rebasing will be something like `git fetch upstream && git rebase upstream/master`.

File: packages/flutter_tools/test/integration.shard/analyze_once_test.dart [hunk 1]
```diff
@@ -100,12 +100,27 @@ void main() {
...
-      exitMessageContains: 'is not a directory',
-      exitCode: 1,
+      statusTextContains: <String>['No issues found!']
+    );
+  });
+
+  testWithoutContext('passing more than one file with errors', () async {
+    final File secondFile = fileSystem.file(fileSystem.path.join(projectPath, 'lib', 'second.dart'))
+      ..createSync(recursive: true)
+      ..writeAsStringSync(r'''import 'package:flutter/material.dart''');
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/analyze_once_test.dart:113]:
```suggestion
      ..writeAsStringSync(r"import 'package:flutter/material.dart");
```

Using double quotes makes the syntax error more obvious.

File: packages/flutter_tools/test/integration.shard/analyze_once_test.dart [hunk 2]
```diff
@@ -100,12 +100,27 @@ void main() {
     );
   });
 
-  // testWithoutContext a specific file outside the current directory
-  testWithoutContext('passing one file throws', () async {
+  testWithoutContext('passing one file works', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/analyze_once_test.dart:103]: These tests look good, can you add two more test cases: `'passing one file with errors are detected'` and `'passing more than one file without errors works'`?

File: packages/flutter_tools/lib/src/commands/analyze_once.dart [hunk 3]
```diff
@@ -48,17 +48,15 @@ class AnalyzeOnce extends AnalyzeBase {
     final String currentDirectory =
         (workingDirectory ?? fileSystem.currentDirectory).path;
 
-    // find directories from argResults.rest
-    final Set<String> directories = Set<String>.of(argResults.rest
+    // find directories or files from argResults.rest
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/analyze_once.dart:51]: does the dartanalyzer support mixing and matching files with directories?

Comment (Jasguerrero) [reply packages/flutter_tools/lib/src/commands/analyze_once.dart:51]: Yeah seems like it support mixing I will also add a test for that case

File: packages/flutter_tools/test/integration.shard/analyze_once_test.dart (hunk 2 above)
Comment (Jasguerrero) [reply packages/flutter_tools/test/integration.shard/analyze_once_test.dart:103]: Sure

**Review by christopherfujino — APPROVED** [review]
LGTM!

**Review by zanderso — APPROVED** [review]
lgtm w/ nit

**Pull request merged**